src/Orleans/Streams/Internal/StreamConsumerExtension.cs

```diff
@@ -231,5 +237,12 @@ internal int DiagCountStreamObservers<T>(StreamId streamId)
                                      .OfType<ObserversCollection<T>>()
                                      .Aggregate(0, (count,o) => count + o.GetObserverCountForStream(streamId));
         }
+
+        internal List<StreamSubscriptionHandleImpl<T>> GetAllStreamHandles<T>()
+        {
+            return allStreamObservers.Values
+                .OfType<ObserversCollection<T>>()
+                .Select(o => o.GetLocalObserver()).ToList();
```

Comment:
I'm not 100% confident all handles should have observers.  Suggest:
 Select(...).Where(h != null).ToList()